src/server/saved_objects/service/lib/repository.js

```diff
@@ -464,6 +464,69 @@ export class SavedObjectsRepository {
     };
   }
 
+  /**
+   * Increases a counter field by one. Creates the document if one doesn't exist for the given id.
+   *
+   * @param {string} type
+   * @param {string} id
+   * @param {string} counterFieldName
+   * @param {object} [options={}]
+   * @property {object} [options.migrationVersion=undefined]
+   * @returns {promise}
+   */
+  async incrementCounter(type, id, counterFieldName, options = {}) {
+    const {
+      migrationVersion,
```

Comment:
@chrisdavies I'd greatly appreciate it if you could just check my usage of your migration stuff. I don't really have an understanding of how it works, I just copied what you did to other methods 😅